tests/Agency/StorePerformanceTest.cpp

```diff
@@ -0,0 +1,651 @@
+                               ", transact failed processing " + json);
+    }
+  }
+
+  void writeAndCheck(std::string const& query) {
+    try {
+      auto r{write(query)};
+      auto applied_all = std::all_of(r.begin(), r.end(), [](auto const& result) {
+        return result == consensus::apply_ret_t::APPLIED;
+      });
+      ASSERT_TRUE(applied_all);
+      if (!applied_all) {
+        throw std::runtime_error("Not all applied");
+      }
+    } catch (std::exception& ex) {
+      throw std::runtime_error(std::string(ex.what()) + " processing " + query);
+    }
+  }
+
+  void writeAndCheck(consensus::query_t const& query) {
+    try {
+      auto r{write(query)};
+      auto applied_all = std::all_of(r.begin(), r.end(), [](auto const& result) {
+        return result == consensus::apply_ret_t::APPLIED;
+      });
+      ASSERT_TRUE(applied_all);
+      if (!applied_all) {
+        throw std::runtime_error("Not all applied");
+      }
+    } catch (std::exception& ex) {
+      throw std::runtime_error(std::string(ex.what()) + " processing " + query->toJson());
+    }
+  }
+
+  void assertEqual(std::shared_ptr<velocypack::Builder> result,
+                   std::string const& expected_result) const {
+    try {
+      auto expected{VPackParser::fromJson(expected_result)};
+      if (!velocypack::NormalizedCompare::equals(result->slice(), expected->slice())) {
+        throw std::runtime_error(
+            result->toJson() + " should have been equal to " + expected->toJson());
+      }
+    } catch (std::exception& ex) {
+      throw std::runtime_error(std::string(ex.what()) + " comparing to " + expected_result);
+    }
+  }
+
+  template <typename TSource>
+  static std::string to_json_object(
+      TSource const& src,
+      std::function<std::string(typename TSource::const_reference)> extractName,
+      std::function<std::string(typename TSource::const_reference)> extractValue) {
+    bool is_first{true};
+    return "{" +
+           std::accumulate(src.begin(), src.end(), std::string(),
+                           [&is_first, extractName,
+                            extractValue](std::string partial, auto const& element) {
+                             if (is_first) {
+                               is_first = false;
+                             } else {
+                               partial += ",";
+                             }
+                             partial += "\"" + extractName(element) +
+                                        "\": " + extractValue(element);
+                             return partial;
+                           }) +
+           "}";
+  }
+
+  template <typename TSource>
+  static std::string to_json_object(TSource const& src) {
+    return to_json_object(
+        src, [](auto const& element) { return element.first; },
+        [](auto const& element) { return element.second; });
+  }
+
+  void threaded_writes(std::function<std::string()> value_generator) {
```

Comment:
Unfortunatly, this measurement is not good for multiple threads. Let me explain why:
1. You measure the points in time at which any thread **finishes** an operation. That means that you do not measure the individual latency of each thread independently with your class. Rather, you measure the time when the first thread finishes, then the time from when the first thread finished to the time when the second finished etc. This is not what we want.
2. Furthermore, even if you correct this, you only measure the complete time a thread took and not the individual latency of each write. I think each thread should have its own `OperationMeasurement` class and then you want to do some reporting for all threads.

Reply:
Let's try that.

Reply:
Oh, sorry I just re-read your comment, it is not correct. Please look at the new parameter ("sequential") on the report() method.

Reply:
I can do the latency of each write, if you think it bears value (I understand they are important for understanding minimums and maximums).

Reply:
This was incorporated.